source/adios2/toolkit/transport/file/FileFlexNVMe.cpp

```diff
@@ -38,22 +40,44 @@ FileFlexNVMe::FileFlexNVMe(helper::Comm const &comm)
 {
 }
 
-FileFlexNVME::~FileFlexNVME() {}
-
-void FileFlexNVME::WaitForOpen() {}
+FileFlexNVMe::~FileFlexNVMe() {}
 
 void FileFlexNVMe::Open(const std::string &name, const Mode openMode,
                         const bool async, const bool directio)
 {
+    m_baseName = name;
 }
 
 void FileFlexNVMe::OpenChain(const std::string &name, Mode openMode,
                              const helper::Comm &chainComm, const bool async,
                              const bool directio)
 {
+    m_baseName = name;
 }
 
-void FileFlexNVME::Write(const char *buffer, size_t size, size_t start) {}
+void FileFlexNVMe::Write(const char *buffer, size_t size, size_t start)
```

Comment:
Looks good, assuming it's fine to have different pools with different object sizes on the same device. If that assumption is incorrect, we probably need to set a shared parameter for all `FileFlexNVMe` instances.